File: backend/src/main/java/com/example/smart/lighting/scenes/with_natural/language/entity/Room.java
package com.example.smart.lighting.scenes.with_natural.language.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

/**
 * Entity representing a room in the smart home.
 *
 * <p>Rooms are containers for {@link Device} entities and provide logical
 * grouping for lighting control. Each room can have multiple devices and
 * tracks which user created it.</p>
 *
 * <p>Room names must be unique across the system.</p>
 *
 * @author Smart Lighting Team
 * @version 1.0
 * @see Device
 * @see User
 */
@Entity
@Table(name = "rooms", schema = "smartlighting")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @OneToMany(mappedBy = "room", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Device> devices = new ArrayList<>();

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: mobile/app/src/main/java/com/smartlighting/mobile/data/model/Room.kt
package com.smartlighting.mobile.data.model

import com.google.gson.annotations.SerializedName

/**
 * Data model representing a room/group with smart lights
 */
data class Room(
    @SerializedName("id")
    val id: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("description")
    val description: String? = null,
    @SerializedName("devices")
    val devices: List<Device>? = null,
    @SerializedName("devicesList")
    val devicesList: List<Device>? = null,
    @SerializedName("createdBy")
    val createdBy: String? = null,
    @SerializedName("createdByName")
    val createdByName: String? = null,
    @SerializedName("createdAt")
    val createdAt: String? = null,
    @SerializedName("updatedAt")
    val updatedAt: String? = null
)


